src/render/posts/does-tdd-work.html.md

```diff
@@ -0,0 +1,228 @@
+
+> The essence of TDD as a design methodology is virtually unstudied, yet scattered early adoption has proceeded based solely on anecdotal evidence.   
+
+> While empirical studies will rarely produce absolute repeatable results, such studies can provide evidence of causal relationships, implying results that will most likely occur in given contexts.  
+
+This to me is a great point. Measurement is difficult and getting fully conclusive results for something like this is almost impossible but we can still use the evidence to help make decisions.
+
+> Survey data reveals that developer opinions about the TDD process improve with TDD experience whereas opinions of test-last programming decrease.  
+
+Much of the data in that paper relies on surveys. These surveys indicated that the more knowledge and experience a developer had with testing and software development in general, the more likely the developer would be of having a better experience with TDD as a process. Interesting but not a major revelation of course. 
+
+Some conclusion the paper arrived at:
+
+>This research has demonstrated that TDD can and is likely to improve some software
+quality aspects at minimal cost over a comparable test-last approach. In particular
+it has shown statistically significant differences in the areas of code complexity,
+size, and testing.
+
+This did appear to be the conclusion based on the data but this is a while back now so it's difficult to say if this is as true today. Also the experiments were on such small groups that it's hard to take this as a fact. To say the results are statistically significant could be a bit of a stretch but it's certainly better than no data.
+
+>This research revealed a number of differences between TDD acceptance and
+efficacy in beginning and mature developers. 
+
+This is just an interesting observation to me. Why do more experienced developers prefer TDD to newer developers? 
+
+A similar work: [Overview of the Test Driven Development
+Research Projects and Experiments ](http://proceedings.informingscience.org/InSITE2012/InSITE12p165-187Bulajic0052.pdf) also looks at varied research papers. This one is slightly newer (2012) and included some more recent research. A significant addition is a study with IBM and Microsoft development teams.
+
+Final conclusion in this study was (Nagappan et al., 2008):
+* Reducing of defect density (IBM 40%, Microsoft 60% - 90%)
+* Increase of time taken to code feature (15% - 35%).
+
+Threats to validity of the study were identified as (Nagappan et al., 2008):
+* Higher motivation of developers that were using TDD methodology.
+* The project developed by using TDD might be easier. 
+
+The conclusion of that paper were less positive than Janzen's as they rightly concluded the results were too varied and sample sizes too small to draw any positive conclusions.
```

Comment:
There is quite a high frequency of the word "conclude/conclusion" in this sentence as well!